web-services/query/src/main/java/datawave/webservice/query/logic/composite/CompositeUserOperations.java

```diff
@@ -45,14 +49,21 @@ public AuthorizationsListBase listEffectiveAuthorizations(Object callerObject) t
             principal.getProxiedUsers().forEach(u -> authMap.put(dn(u.getDn()), new HashSet<>(u.getAuths())));
         }
         for (UserOperations ops : userOperations) {
-            AuthorizationsListBase remoteAuths = ops.listEffectiveAuthorizations(callerObject);
-            AuthorizationsListBase.SubjectIssuerDNPair userDn = new AuthorizationsListBase.SubjectIssuerDNPair(remoteAuths.getUserDn(),
-                            remoteAuths.getIssuerDn());
-            authMap.put(userDn, Sets.union(authMap.containsKey(userDn) ? authMap.get(userDn) : Collections.emptySet(), remoteAuths.getAllAuths()));
-            Map<AuthorizationsListBase.SubjectIssuerDNPair,Set<String>> remoteAuthMap = remoteAuths.getAuths();
-            for (Map.Entry<AuthorizationsListBase.SubjectIssuerDNPair,Set<String>> entry : remoteAuthMap.entrySet()) {
-                AuthorizationsListBase.SubjectIssuerDNPair dn = entry.getKey();
-                authMap.put(dn, Sets.union(authMap.containsKey(dn) ? authMap.get(dn) : Collections.emptySet(), entry.getValue()));
+            try {
+                AuthorizationsListBase remoteAuths = ops.listEffectiveAuthorizations(callerObject);
+                AuthorizationsListBase.SubjectIssuerDNPair userDn = new AuthorizationsListBase.SubjectIssuerDNPair(remoteAuths.getUserDn(),
+                                remoteAuths.getIssuerDn());
+                authMap.put(userDn, Sets.union(authMap.containsKey(userDn) ? authMap.get(userDn) : Collections.emptySet(), remoteAuths.getAllAuths()));
+                Map<AuthorizationsListBase.SubjectIssuerDNPair,Set<String>> remoteAuthMap = remoteAuths.getAuths();
+                for (Map.Entry<AuthorizationsListBase.SubjectIssuerDNPair,Set<String>> entry : remoteAuthMap.entrySet()) {
+                    AuthorizationsListBase.SubjectIssuerDNPair dn = entry.getKey();
+                    authMap.put(dn, Sets.union(authMap.containsKey(dn) ? authMap.get(dn) : Collections.emptySet(), entry.getValue()));
+                }
+            } catch (AuthorizationException e) {
+                // ignore the exception if shortCircuitExecution is specified as we may never even call that remote logic
+                if (!shortCircuitExecution) {
```

Comment:
I'm curious if we could throw an AuthorizationException for a non-remote user operations object. Should this be
```java
if (!shortCircuitExecution && op instanceof RemoteUserOperations) {
```

Reply:
I don't think it needs to be exclusively a RemoteUserOperations.  We could have other mechanisms or implementations that need the same handling from the composite user operations standpoint.